kafka/README.adoc

```diff
@@ -0,0 +1,159 @@
+and other general information.
+
+
+== Prerequisites
+
+The example application requires a running Kafka instance. For simplicity, you can launch the Kafka instance using the docker-compose.yaml.
+
+----
+$ cd conf && docker-compose up
+----
+
+Next : uncomment the The section Kafka instance without Authentication in `src/main/resources/application.properties` and set :
+----
+camel.component.kafka.brokers=localhost:9092
+----
+
+If you prefer to use a different Kafka instance uncomment and adjust the corresponding commented section in `src/main/resources/application.properties`.
+
+- The section Kafka instance without Authentication if no Authentication required.
+- The section Kafka instance with SASL Plain is using SASL.
+- The section Kafka instance with SASL Oauth Bearer if using Oauth Bearer.
+
+== Start in Development mode
+
+Run the application in development mode.
+
+[source,shell]
+----
+$ mvn clean compile quarkus:dev
+----
+
+The above command compiles the project, starts the application and lets the Quarkus tooling watch for changes in your
+workspace. Any modifications in your project will automatically take effect in the running application.
+
+TIP: Please refer to the Development mode section of
+https://camel.apache.org/camel-quarkus/latest/first-steps.html#_development_mode[Camel Quarkus User guide] for more details.
+
+You should start to see some log messages appearing on the console.
+
+Every 10 seconds the timer component triggers the generation of random Message and send it to the Kafka topic `Test`.
+
+[source,shell]
+----
+[FromTimer2Kafka] (Camel (camel-1) thread #2 - KafkaProducer[test]) Message sent correctly sent to the topic! : "Message #1"
+----
+
+Next a Kafka consumer reads the messages and put them in a seda queue.
+
+[source,shell]
+----
+[FromKafka2Seda] (Camel (camel-1) thread #0 - KafkaConsumer[test]) Received : "Message #1"
+----
+
+Next pull a message from the queue :
+[source,shell]
+----
+$ curl -X GET http://0.0.0.0:8080/example
+----
+
+
+=== Package and run the application
+
+Once you are done with developing you may want to package and run the application.
+
+TIP: Find more details about the JVM mode and Native mode in the Package and run section of
+https://camel.apache.org/camel-quarkus/latest/first-steps.html#_package_and_run_the_application[Camel Quarkus User guide]
+
+==== JVM mode
+
+[source,shell]
+----
+$ mvn clean package -DskipTests
+$ java -jar target/quarkus-app/quarkus-run.jar
+----
+
+==== Native mode
+
+IMPORTANT: Native mode requires having GraalVM and other tools installed. Please check the Prerequisites section
+of https://camel.apache.org/camel-quarkus/latest/first-steps.html#_prerequisites[Camel Quarkus User guide].
+
+To prepare a native executable using GraalVM, run the following command:
+
+[source,shell]
+----
+$ mvn clean package -DskipTests -Pnative
+$ ./target/*-runner
+----
+
+==== Deploying to Kubernetes
```

Comment:
Just a suggestion, we could just have a 'Deploying to OpenShift section' since we're not so focused on vanilla k8s in the product. I did that in the `file-bindy-ftp` example README.